File: stacks_and_queues/number_of_recent_calls.js
var RecentCounter = function() {
    this.queue = [];
};

/**
 * @param {number} t
 * @returns {number}
 */
RecentCounter.prototype.ping = function(t) {
    while (this.queue.length && this.queue[0] < t - 3000)
    {
        this.queue.shift();
    }

    this.queue.push(t);
    return this.queue.length;
};

File: stacks_and_queues/number_of_recent_calls.java
class RecentCounter
{
    Queue<Integer> queue;

    public RecentCounter()
    {
        queue = new LinkedList<>();
    }

    public int ping(int t)
    {
        while (!queue.isEmpty() && queue.peek() < t - 3000)
        {
            queue.poll();
        }

        queue.offer(t);
        return queue.size();
    }
}